samcli/lib/bootstrap/companion_stack/companion_stack_manager.py

```diff
@@ -122,16 +123,16 @@ def update_companion_stack(self) -> None:
             self._cfn_client.update_stack(
                 StackName=stack_name, TemplateURL=template_url, Capabilities=["CAPABILITY_AUTO_EXPAND"]
             )
-            waiter = self._cfn_client.get_waiter("stack_update_complete")
+            update_waiter = self._cfn_client.get_waiter("stack_update_complete")
+            update_waiter.wait(StackName=stack_name, WaiterConfig=self._update_stack_waiter_config)
         else:
             self._cfn_client.create_stack(
                 StackName=stack_name, TemplateURL=template_url, Capabilities=["CAPABILITY_AUTO_EXPAND"]
             )
-            waiter = self._cfn_client.get_waiter("stack_create_complete")
+            create_waiter = self._cfn_client.get_waiter("stack_create_complete")
+            create_waiter.wait(StackName=stack_name, WaiterConfig=self._update_stack_waiter_config)
```

Comment:
Even though lines 127 and 133 are identical, it is easier for type checking, because `StackUpdateCompleteWaiter` and `StackCreateCompleteWaiter` do not share the same `wait` method.